src/Entity/EntityId.php

```diff
@@ -179,4 +191,10 @@ public function equals( $target ) {
 			&& $target->serialization === $this->serialization;
 	}
 
+	protected function getRepositoryNameAndLocalPart( $serialization ) {
+		list( $repoName, $prefixRemainder, $localId ) = self::getSerializationParts( $serialization );
```

Comment:
This bypasses validation. That seems odd, but if this is really desired, the contract documentation should make this clear.

Reply:
This indeed was a point. Namely, it is meant to be used in unserialize methods. As it has been bypassing validation for the whole "serialization", I went for the same behaviour here. I don't claim it is a right thing to do, but OTOH it was kind of something Stas was talking about :) (do not validate stuff you know is right).